contracts/libraries/OFTTransportAdapter.sol

```diff
@@ -0,0 +1,111 @@
+     */
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint32 public immutable DST_EID;
+
+    /**
+     * @notice intiailizes the OFTTransportAdapter contract.
+     * @param _oftDstEid the endpoint ID that OFT protocol will transfer funds to.
+     * @param _feeCap a fee cap we check against before sending a message with value to OFTMessenger as fees.
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(uint32 _oftDstEid, uint256 _feeCap) {
+        DST_EID = _oftDstEid;
+        FEE_CAP = _feeCap;
+    }
+
+    /**
+     * @notice transfer token to the other dstEId (e.g. chain) via OFT messaging protocol
+     * @dev the caller has to provide both _token and _messenger. The caller is responsible for knowing the correct _messenger
+     * @param _token token we're sending on current chain.
+     * @param _messenger corresponding OFT messenger on current chain.
+     * @param _to address to receive a trasnfer on the destination chain.
+     * @param _amount amount to send.
+     */
+    function _transferViaOFT(
+        IERC20 _token,
+        IOFT _messenger,
+        address _to,
+        uint256 _amount
+    ) internal {
+        bytes32 to = _to.toBytes32();
+
+        SendParam memory sendParam = SendParam(
+            DST_EID,
+            to,
+            /**
+             * _amount, _amount here specify `amountLD` and `minAmountLD`. These 2 have a subtle relationship
+             * OFT "removes dust" on .send, which doesn't affect tokens like `USDT` because `USDT.decimals() = USDT_IOFT.sharedDecimals()`.
+             * However, for other tokens we need to make sure that we're passing the correct `amount` to this function.
+             * In order for OFT to remove zero dust, we need to make sure that last `token.decimals() - token_IOFT.sharedDecimals()` digits
+             * in our amount are set to 0.
+             * That said, setting both these vars to `_amount` protects us from dust subtraction on `OFT` contract side, which will revert
+             * if we pass amount with some dust. It's important for our HubPool accounting to keep this behavior, meaning have
+             * `sentAmount == receivedAmount` always
+             */
+            _amount,
+            _amount,
+            /**
+             * EMPTY_MSG_BYTES, EMPTY_MSG_BYTES, EMPTY_MSG_BYTES here specify `extraOptions`, `composeMsg` and `oftCmd`.
+             * These can be set to empty bytes arrays for the purposes of sending a simple cross-chain transfer.
+             */
+            EMPTY_MSG_BYTES,
+            EMPTY_MSG_BYTES,
+            EMPTY_MSG_BYTES
+        );
+
+        // `false` in the 2nd param here refers to `bool _payInLzToken`. We will pay in native token, so set to `false`
+        MessagingFee memory fee = _messenger.quoteSend(sendParam, false);
+        require(fee.nativeFee <= FEE_CAP, "OFT FEE_CAP exceeded");
```

Comment:
nit prefer using custom [Errors](https://github.com/across-protocol/contracts/blob/555475cdee6109afc85065ca415c740d7f97b992/contracts/interfaces/V3SpokePoolInterface.sol#L326) to save gas costs